### PR TITLE
Add Validate Operator Bundle check to CraneEngine

### PR DESCRIPTION
**Pull request by bcrochet**
The Validate Operator Bundle will now look for a bundle path, instead of a bundle image.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

**Review by jomkz — APPROVED** [review]
/lgtm

**Review by acornett21 — APPROVED** [review]

**Comment by bcrochet** [comment]
I will validate that this does not require privilege before merging.

/hold

**Comment by bcrochet** [comment]
/hold cancel

This is passing without privilege now.

File: certification/engine/operatorsdk.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+	cmdArgs = append(cmdArgs, image)
+
+	cmd := exec.Command("operator-sdk", cmdArgs...)
+	log.Trace("running scorecard with the following invocation", cmd.Args)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err = cmd.Run()
+	if err != nil {
```
Comment (samira-barouti) [inline certification/engine/operatorsdk.go:62]: You could merge these two lines into one.

File: certification/engine/operatorsdk.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		// We also conclude/assume that anything being in stderr would indicate an error in the
+		// check execution itself.
+		if stderr.Len() != 0 {
+			log.Error("stderr: ", stdout.String())
+			return nil, fmt.Errorf("%w: %s", errors.ErrOperatorSdkScorecardFailed, err)
+		}
+	}
+
+	err = o.writeScorecardFile(opts.ResultFile, stdout.String())
+	if err != nil {
```
Comment (samira-barouti) [inline certification/engine/operatorsdk.go:78]: same goes here

File: certification/engine/operatorsdk.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		log.Error("unable to copy result to artifacts directory: ", err)
+		return nil, err
+	}
+
+	var scorecardData cli.OperatorSdkScorecardReport
+	err = json.Unmarshal(stdout.Bytes(), &scorecardData)
+	if err != nil {
```
Comment (samira-barouti) [inline certification/engine/operatorsdk.go:85]: same goes here

File: certification/engine/operatorsdk.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+			bundleValidateData.Passed = false
+		}
+	}
+	bundleValidateData.Stdout = stdout.String()
+	bundleValidateData.Stderr = stderr.String()
+
+	return &bundleValidateData, nil
+}
+
+func (o operatorSdkEngine) writeScorecardFile(resultFile, stdout string) error {
```
Comment (samira-barouti) [inline certification/engine/operatorsdk.go:141]: This could be moved to `util/file` for future reuse.

File: certification/engine/operatorsdk.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+			  suite: olm
+			  test: olm-bundle-validation-test
+`
+
+	tempConfigFile, err := os.CreateTemp("", "scorecard-test-config-*.yaml")
+	if err != nil {
+		log.Error("could not create temp config file", err)
+		return "", err
+	}
+	_, err = tempConfigFile.WriteString(configTemplate)
```
Comment (samira-barouti) [inline certification/engine/operatorsdk.go:180]: I am wondering If the tempConfigFile has been created successfully, but the function fails to run `WriteString`, do we want to return tempConfigFile as non-nil value along with a non-nil error?

**Review by samira-barouti — COMMENTED** [review]
A few nits and a question regarding the error handling

File: certification/engine/operatorsdk.go (hunk 5 above)
Comment (samira-barouti) [reply certification/engine/operatorsdk.go:180]: nvm; Found out how the caller is making use of both return values.

**Review by acornett21 — APPROVED** [review]

File: certification/engine/operatorsdk.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+	if opts.ServiceAccount != "" {
+		cmdArgs = append(cmdArgs, "--service-account", opts.ServiceAccount)
+	}
+
+	configFile, err := createScorecardConfigFile()
+	if err != nil {
+		log.Error("could not create scorecard config file", err)
+		return nil, err
+	}
+	defer os.Remove(configFile)
```
Comment (samira-barouti) [inline certification/engine/operatorsdk.go:50]: This defer statement could be moved to line 46 to ensure the configFile is removed in the case of errors.

**Review by samira-barouti — APPROVED** [review]
/lgtm

**Pull request merged**